File: Problem1/R/soln.R
# Solution to Problem 1

upper_bound <- 1000
divisors <- c(3, 5)
multiples <- vector()

N <- length(divisors)

for (i in 1:N)
{
    div <- divisors[i]
    
    mult <- div

    i <- 1

    while (mult < upper_bound)
    {
        
        multiples <- c(multiples, mult)
        
        i <- i + 1 

        mult <- div*i

    }

}

unique_multiples <- unique(multiples)

mult_sum <- sum(unique_multiples)

print(mult_sum)

File: Problem1/Python/soln.py
# Solution to problem 1

import numpy as np

upper_bound = 1000
divisors = set([3, 5])
multiples = set()

for div in divisors:

    i = 1

    mult = div * i

    while mult < upper_bound:

        multiples.add(mult)

        i += 1

        mult = div * i

sum_mult = sum(multiples)

print sum_mult
